addons/web/static/src/legacy/js/views/kanban/kanban_record.js

```diff
@@ -610,31 +610,35 @@ var KanbanRecord = Widget.extend(WidgetAdapterMixin, {
      * @returns {Object} transformed record data
      */
     _transformRecord: function (recordData) {
-        var self = this;
-        var new_record = {};
-        _.each(this.state.getFieldNames(), function (name) {
-            var value = recordData[name];
-            var r = _.clone(self.fields[name] || {});
+        const new_record = Object.create(null);
+        for (const name of this.state.getFieldNames()) {
+            const value = recordData[name];
+            const r = new_record[name] = _.clone(this.fields[name] || {});
 
-            if ((r.type === 'date' || r.type === 'datetime') && value) {
-                r.raw_value = value.toDate();
-            } else if (r.type === 'one2many' || r.type === 'many2many') {
+            r.raw_value = value;
+            switch (r.type) {
+            case 'date': case 'datetime':
+                r.raw_value = value && value.toDate();
+                break;
+            case 'one2many': case 'many2many':
                 r.raw_value = value.count ? value.res_ids : [];
-            } else if (r.type === 'many2one') {
+                break;
+            case 'many2one':
                 r.raw_value = value && value.res_id || false;
-            } else {
-                r.raw_value = value;
+                break;
             }
 
-            if (r.type) {
-                var formatter = field_utils.format[r.type];
-                r.value = formatter(value, self.fields[name], recordData, self.state);
-            } else {
+            switch (r.type) {
+            case 'html': // or should this be in `format` instead of _.identity?
```

Comment:
I think it should, indeed

Reply:
One problem with that is a `Markup` is a `String` but not a `string`. There might be contexts where the formatter is used and expects a string output there. I've had this issue when I tried to make qweb template return `Markup` objects to match Python, and ended up abandoning this idea: string typechecking is mostly done through `typeof <x> === 'string'`, which `String` and `Markup` don't match (their primitive type is object).

Reply:
I still think it should, but we could do this in the new code only (our new versions of formatters)

Reply:
Yeah in the owl formatters it's probably more of an option, less risk of bleed and easier to find compatibity issues, especially around putting HTML content in jQuery which *really* doesn't like boxed strings.